c++/src/kj/compat/http.c++

```diff
@@ -5785,14 +5880,13 @@ private:
     });
   }
 
-  static kj::Promise<ConnectResponse> attachCounter(
-      kj::Promise<ConnectResponse>&& promise,
+  static ConnectRequest attachCounter(
+      ConnectRequest&& request,
       ConnectionCounter&& counter) {
-    return promise.then([counter = kj::mv(counter)](HttpClient::ConnectResponse&& response)
-        mutable -> HttpClient::ConnectResponse {
-      response.connection = response.connection.attach(kj::mv(counter));
-      return kj::mv(response);
-    });
+    // TODO(someday): Should the counter also be attached to the status promise?
```

Comment:
If we bring back `errorBody` is this still true? If the status indicates rejected, the `errorBody` effectively replaces the `connection`, which may be dropped. If we don't attach the counter to `errorBody` (by passing a reference into the promise continuation and attaching to `errorBody` if given), then we risk decrementing to counter prematurely. Is that ok?

Reply:
Ah, interesting. I guess the fully correct thing to do would indeed be to refcount and attach it to both. However, for our use case, I don't think counting error bodies is really needed. In Workers, we don't plan to expose CONNECT directly to applications. We do intend to use CONNECT under the hood to implement TCP connections. However, an error response to a `CONNECT` request would be a response from our internal proxy. We would not expose this response to the application anyway, and we'd always consume said error proactively.

So it _is_ important to count successfully-established connections against the connection limit, but it is _not_ important to count error responses, at least for our use case.

Reply:
Yep. I think going back to your point about making sure it's right for general purpose (and not just our internal use case), I think I'll go ahead and make ConnectionCounter refcounted since we really can't be certain if someone may want to count error responses (even if we don't).